kubernetes.go

```diff
@@ -343,26 +409,26 @@ func provisionedByAwsEbs(pvc *corev1.PersistentVolumeClaim) bool {
 	return false
 }
 
-func processPersistentVolumeClaim(pvc *corev1.PersistentVolumeClaim) (string, map[string]string, error) {
+func processPersistentVolumeClaim(pvc *corev1.PersistentVolumeClaim) (string, map[string]string, *corev1.PersistentVolume, error) {
 	tags := buildTags(pvc)
 
 	log.WithFields(log.Fields{"namespace": pvc.GetNamespace(), "pvc": pvc.GetName(), "tags": tags}).Debugln("PVC Tags")
 
 	pv, err := k8sClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
 	if err != nil {
 		log.WithFields(log.Fields{"namespace": pvc.GetNamespace(), "pvc": pvc.GetName()}).Errorln("Get PV from kubernetes cluster error:", err)
-		return "", nil, err
+		return "", nil, pv, err
 	}
 
 	var volumeID string
 	annotations := pvc.GetAnnotations()
 	if annotations == nil {
 		log.Errorf("cannot get PVC annotations")
-		return "", nil, errors.New("cannot get PVC annotations")
+		return "", nil, pv, errors.New("cannot get PVC annotations")
```

Comment:
Let's be consistent and return nil for the PV in the event of errors like it does for the pvc